### PR TITLE
[Enhancement][QoL] Add option to adjust shop overlay opacity

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes?
- see title
- option is only up to 90%

## Why am I doing these changes?
- some players are still not happy with how dark/light the current opacity is
- adding option instead would allow users to customize their experience

## What did change?
- new Shop Overlay Opacity setting

### Screenshots/Videos
- Menu
Screenshot 2024-07-11 at 11 52 25 PM

- 90%
Screenshot 2024-07-11 at 11 48 44 PM

- 80% (default)
Screenshot 2024-07-11 at 11 46 35 PM

- 70%
Screenshot 2024-07-11 at 11 49 10 PM

- 60%
Screenshot 2024-07-11 at 11 49 42 PM

- 50%
Screenshot 2024-07-11 at 11 50 07 PM

- 40%
Screenshot 2024-07-11 at 11 50 36 PM

- 30%
Screenshot 2024-07-11 at 11 50 55 PM

- 20%
Screenshot 2024-07-11 at 11 51 19 PM

- 10%
Screenshot 2024-07-11 at 11 51 39 PM

## How to test the changes?
- get to shop phase
- change overlay value and check the change

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/fr/settings.ts [hunk 1]
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Montrer titre de la musique",
+  "none": "None",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
Comment (Adri1) [inline src/locales/fr/settings.ts:100]:
```suggestion
  "none": "Aucune",
  "shopOverlayOpacity": "Opacité boutique"
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/zh_CN/settings.ts [hunk 2]
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "控制器",
   "gamepadSupport": "手柄支持",
   "showBgmBar": "显示音乐名称",
+  "none": "None",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
Comment (40chyan) [inline src/locales/zh_CN/settings.ts:100]:
```suggestion
  "none": "无",
  "shopOverlayOpacity": "商店显示不透明度"
```

File: src/locales/fr/settings.ts [hunk 3]
```diff
@@ -96,4 +96,6 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Montrer titre de la musique",
```
Comment (Adri1) [inline src/locales/fr/settings.ts:98]:
```suggestion
  "showBgmBar": "Titre de la musique",
```

**Review by Dakurei — APPROVED** [review]
Tested locally, no particular errors encountered during testing 👍

**Comment by torranx** [comment]
Updated values to be 20 30 40 50 60 70 80 90

- Removed `0` as this would not make sense to have.
- Removed `10` as well as this is visually the same with `0`

**Review by Dakurei — APPROVED** [review]

File: src/locales/pt_BR/settings.ts [hunk 4]
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controle",
   "gamepadSupport": "Suporte para Controle",
   "showBgmBar": "Show Music Names",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
Comment (josericardo-fo) [inline src/locales/pt_BR/settings.ts:99]:
```suggestion
  "shopOverlayOpacity": "Opacidade da Loja"
```

File: src/locales/ko/settings.ts [hunk 5]
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "컨트롤러",
   "gamepadSupport": "게임패드 지원",
   "showBgmBar": "BGM 제목 보여주기",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
Comment (EnochG1) [inline src/locales/ko/settings.ts:99]:
```suggestion
  "shopOverlayOpacity": "상점 오버레이 투명도"
```

**Review by EnochG1 — COMMENTED** [review]
Korean translation

File: src/locales/es/settings.ts [hunk 6]
```diff
@@ -96,4 +96,5 @@ export const settings: SimpleTranslationEntries = {
   "controller": "Controller",
   "gamepadSupport": "Gamepad Support",
   "showBgmBar": "Show Music Names",
+  "shopOverlayOpacity": "Shop Overlay Opacity"
```
Comment (InnocentGameDev) [inline src/locales/es/settings.ts:99]:
```suggestion
  "shopOverlayOpacity": "Opacidad de la fase de compra"
```

**Review by InnocentGameDev — COMMENTED** [review]
Es localization

**Review by Tempo-anon — COMMENTED** [review]
Do you mind adding screenshots for how the same shop would look like at each opacity?

**Comment by torranx** [comment]
> Do you mind adding screenshots for how the same shop would look like at each opacity?

updated PR description

**Pull request merged**